test/test_nn.py

```diff
@@ -10260,6 +10260,12 @@ def test_hardsigmoid_grad(self, device):
         self.assertTrue(gradcheck(F.hardsigmoid, (inputs,)))
 
 
+    def test_hardswish_grad(self, device):
+        inputs = (torch.randn(4, 16, 16, device=device) - 0.5) * 10
+        inputs.requires_grad = True
+        self.assertTrue(gradcheck(F.hardswish, (inputs,)))
```

Comment:
How does this work? Where is the reference output provided here?

Reply:
this is a nice util function, it checks that the gradient computed analytically by perturbing the input matches the gradient defined by the backwards pass.  It's used to test other backwards passes in this file.

https://github.com/pytorch/pytorch/blob/73a36a47a5d371f34c750b442a4258fe5fc97504/torch/autograd/gradcheck.py#L198 has more context